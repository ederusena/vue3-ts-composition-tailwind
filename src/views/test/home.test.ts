import { describe, it } from "vitest";
import { shallowMount } from "@vue/test-utils";

import Home from "../HomeView.vue";

describe("HomePage.vue", () => {
  const wrapper = shallowMount(Home);

  it("should render correct contents", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render H1 that contains Home Page text", () => {
    expect(wrapper.find("h1").text()).toBe("Home Page");
  });

  it("should the counter element start with zero", () => {
    expect(wrapper.find(".counter").text()).toBe("0");
  });

  it("should the counter element increment by one when the button is clicked", async () => {
    await wrapper.find(".increment").trigger("click");
    expect(wrapper.find(".counter").text()).toBe("1");
  });

  it("should the counter element decrement by one when the button is clicked", async () => {
    await wrapper.find(".decrement").trigger("click");
    expect(wrapper.find(".counter").text()).toBe("0");
  });

  it("should the counter element increment by two when the button is clicked", async () => {
    await wrapper.find(".double-increment").trigger("click");
    expect(wrapper.find(".counter").text()).toBe("2");
  });

  it("should the counter element decrement by two when the button is clicked", async () => {
    await wrapper.find(".double-decrement").trigger("click");
    expect(wrapper.find(".counter").text()).toBe("0");
  });
});
