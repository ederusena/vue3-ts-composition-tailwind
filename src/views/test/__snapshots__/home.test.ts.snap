// Vitest Snapshot v1

exports[`HomePage.vue > should render correct contents 1`] = `
"<div class=\\"container flex flex-col mx-auto justify-center bg-gray-100\\">
  <h1 class=\\"mx-auto text-2xl p-4\\">Home Page</h1>
  <div class=\\"flex justify-center gap-2\\"><button class=\\"btn double-decrement\\">--</button><button class=\\"btn decrement\\">-</button><span class=\\"counter text-3xl font-bold\\">0</span><button class=\\"btn increment\\">+</button><button class=\\"btn double-increment\\">++</button></div>
  <div class=\\"flex flex-col mx-auto p-4\\">
    <div class=\\"flex flex-col\\">
      <h4>Edit counter title: My Counter</h4><input type=\\"text\\" class=\\"bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500\\">
    </div>
    <p>This counter is even</p>
  </div>
</div>"
`;
